src/native/libs/System.Native/CMakeLists.txt

```
@@ -41,6 +41,15 @@ else()
     )
 endif()
 
+if (CLR_CMAKE_TARGET_APPLE OR CLR_CMAKE_TARGET_ANDROID OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
+    list (APPEND NATIVE_SOURCES
+        pal_mutex_unsupported.c)
```

```suggestion
        pal_crossprocessmutex_unsupported.c)
```
This is not an ordinary mutex...